lib/capistrano/slack_notify.rb

```diff
@@ -91,8 +91,12 @@ def destination
       fetch(:slack_destination, stage)
     end
 
+    def repository
+      fetch(:repository, 'origin')
+    end
+
     def revision
-      @revision ||= `git rev-parse #{branch}`.chomp
+      @revision ||= `git ls-remote #{repository} #{branch}`.chomp
```

Comment:
When I run this locally, I get soemthing like:

```text
aa04140db6643ac65152cba813085815dfeb1141	refs/heads/master
```

It looks like this will need more modification! `.split(" ").first`